modules/social_features/social_group/social_group.module

```diff
@@ -41,10 +44,51 @@ function social_group_form_group_public_group_edit_form_alter(&$form, FormStateI
   $group = $form_state->getFormObject()->getEntity();
   $form['group_type'] = SocialGroupAddForm::create(\Drupal::getContainer())->getGroupTypeElement();
   $form['group_type']['#default_value'] = $group->bundle();
-  $form['group_type']['#disabled'] = TRUE;
+  // If user doesn't have permission to change group types disable it.
+  if (!social_group_group_type_permission_check()) {
+    $form['group_type']['#disabled'] = TRUE;
+  }
+  else {
+    $form['group_type']['#prefix'] = '<div id="group-type-result"></div>';
+    $group_type_element['#ajax'] = [
+      'callback' => '_social_group_inform_group_type_selection',
+      'effect' => 'fade',
+      'event' => 'change',
+    ];
+  }
   $form['#group_children']['group_type'] = 'group_content';
 }
 
+/**
+ * Check if a User is able to edit a Group's GroupType.
+ *
+ * @return bool
+ *   TRUE if a user may edit a existings groups group type.
+ */
+function social_group_group_type_permission_check() {
+  $user = \Drupal::currentUser();
+  // Get the Group object from the route.
+  $group = Drupal::routeMatch()->getParameter('group');
+
+  // Check if we are editing a group. Sure to be sure.
+  if ($group instanceof GroupInterface) {
+    // Only for user 1 and users with the edit group types permission we can
+    // edit a group type.
+    if ($user->id() != 1) {
```

Comment:
@ronaldtebrake shouldn't we just check for the permission? Why are we checking on user ID here?